src/plugins/mailer/builder.ts

```diff
@@ -0,0 +1,169 @@
+import { TFunction } from 'i18next';
+
+import { DEFAULT_LANG } from '@graasp/translations';
+
+import { GRAASP_LANDING_PAGE_ORIGIN } from '../../utils/constants';
+import i18next from './i18n';
+import { MAIL } from './langs/constants';
+import { Mail } from './service';
+
+type CssStyles = { [key: string]: string };
+
+export class MailBuilder {
+  private subject: string;
+  private plainText = '';
+  private htmlText = '';
+  private buttonText: string;
+  private translationVariables: { [key: string]: string };
+  private language: string = DEFAULT_LANG;
+
+  private t: TFunction;
+
+  constructor({
+    subject,
+    translationVariables,
+    lang,
+  }: {
+    subject: string;
+    translationVariables: { [key: string]: string };
+    lang?: string;
+  }) {
+    this.subject = subject;
+    this.translationVariables = translationVariables;
+
+    if (lang) {
+      this.language = lang;
+    }
+
+    i18next.changeLanguage(this.language);
+    this.t = i18next.t;
+  }
+
+  public addText(stringId: string): MailBuilder {
+    const unescapedText = this.t(stringId, {
+      ...this.translationVariables,
+      interpolation: { escapeValue: false },
+    });
+    this.plainText += `\n${unescapedText}\n`;
+
+    const escapedText = this.t(stringId, this.translationVariables);
+    this.htmlText += this.buildText(escapedText);
+
+    return this;
+  }
+
+  public addButton(buttonText: string, callToActionLink: string): MailBuilder {
+    this.plainText += `\n${callToActionLink}\n`;
+
+    this.htmlText += this.buildButton(
+      callToActionLink,
+      this.t(buttonText, this.translationVariables),
+    );
+
+    // keep this text for user agreement
+    this.buttonText = buttonText;
+
+    return this;
+  }
+
+  public includeUserAgreement(): MailBuilder {
```

Comment:
I'm wondering if we should not call it `addUserAgreement` to be more coherent with the other function names? (I know that was my proposition ^^)

Maybe we would like to keep `includeUserAgreement`, but in this case, what about including once, even if we call this function twice? I don't know, I'm just thinking about that, it's not mandatory, but it could reflect that you include once.